src/test_typing_extensions.py

```diff
@@ -850,6 +853,33 @@ def d(): pass
             isinstance(cell.cell_contents, deprecated) for cell in d.__closure__
         ))
 
+    def test_inspect(self):
+        @deprecated("depr")
+        def sync():
+            pass
+
+        @deprecated("depr")
+        async def coro():
+            pass
+
+        class Cls:
+            @deprecated("depr")
+            def sync(self):
+                pass
+
+            @deprecated("depr")
+            async def coro(self):
+                pass
+
+        self.assertFalse(inspect.iscoroutinefunction(sync))
+        self.assertFalse(inspect.iscoroutinefunction(Cls.sync))
+        if sys.version_info >= (3, 12) and not TYPING_3_13_BETA:
+            self.assertTrue(inspect.iscoroutinefunction(coro))
+            self.assertTrue(inspect.iscoroutinefunction(Cls.coro))
+        else:
+            self.assertFalse(inspect.iscoroutinefunction(coro))
+            self.assertFalse(inspect.iscoroutinefunction(Cls.coro))
```

Comment:
I'd prefer to explicitly skip the test if we can't get the behaviour we'd like on some versions, rather than have the test assert different things on different Python versions. I don't think it's really desirable behaviour for `inspect.iscoroutinefunction()` to return `False` for `@deprecated` functions on earlier Python versions (we just can't really fix it), and I'm not sure we should be asserting undesirable behaviour.